core/shared/src/main/scala/zio/FiberRef.scala

```diff
@@ -185,7 +185,7 @@ trait FiberRef[A] extends Serializable { self =>
    * value when the scope is closed.
    */
   final def locallyScopedWith(f: A => A)(implicit trace: Trace): ZIO[Scope, Nothing, Unit] =
-    getWith(a => locallyScoped(f(a))(trace))
+    ZIO.acquireRelease(get.flatMap(a => set(f(a)).as(a)))(set).unit
```

Comment:
Can't we use `getAndSet` here?

Reply:
Hum no but we can use `modify`. I'll update the implementation